modules/smtp/modules.php

```diff
@@ -117,13 +117,62 @@ public function process() {
 /**
  * @subpackage smtp/handler
  */
-class Hm_Handler_load_smtp_is_imap_forward extends Hm_Handler_Module
+class Hm_Handler_load_smtp_is_imap_forward_as_attachment extends Hm_Handler_Module
 {
     public function process() {
         if (!$this->module_is_supported('imap')) {
             return;
         }
 
+        if (array_key_exists('forward_as_attachment', $this->request->get)) {
+            $path = explode('_', $this->request->get['list_path']);
+            $imap = Hm_IMAP_List::connect($path[1]);
+            if ($imap && $imap->select_mailbox(hex2bin($path[2]))) {
+                $msg_struct = $imap->get_message_structure($this->request->get['uid']);
+                list($part, $msg_text) = $imap->get_first_message_part($this->request->get['uid'], 'text', 'plain', $msg_struct);
+                $msg_header = $imap->get_message_headers($this->request->get['uid']);
+                if (!array_key_exists('From', $msg_header) || count($msg_header) == 0) {
+                    return;
+                }
+                $content = $imap->get_message_content($this->request->get['uid'], 0, false, false);
+
+                # Attachment Download
+                $attached_files = [];
+                $this->session->set('uploaded_files', array());
+
+                $file_dir = $this->config->get('attachment_dir') . DIRECTORY_SEPARATOR . md5($this->session->get('username', false)) . DIRECTORY_SEPARATOR;
+                if (!is_dir($file_dir)) {
+                    mkdir($file_dir);
+                }
+                $name = $msg_header['subject'] . '.eml';
+                $file_path = $file_dir . $name;
+                $attached_files[$this->request->get['uid']][] = array(
+                    'name' => $name,
+                    'type' => 'message/rfc822',
+                    'size' => strlen($content),
+                    'tmp_name' => $file_path,
+                    'filename' => $file_path,
+                    'basename' => $msg_header['subject']
+                );
+                
+                $content = Hm_Crypt::ciphertext($content, Hm_Request_Key::generate());
```

Comment:
Shouldn't we use the current generated request key instead of generating a new one?